testcases/OpTestKexec.py

```diff
@@ -0,0 +1,160 @@
+        res = self.c.run_command("cat /etc/os-release")
+        if "Ubuntu" in (res[0] or res[1]):
+            self.distro = "Ubuntu"
+        elif 'Red Hat' in (res[0] or res[1]):
+            self.distro = 'RHEL'
+        elif 'SLES' in (res[0] or res[1]):
+            self.distro = 'SLES'
+        else:
+            raise self.skipTest("Test currently supported only on Ubuntu, SLES and RHEL")
+ 
+    def get_raw_pty_console(self,cmd):
+        """
+        This function executes the command in raw_pty console 
+        """
+        try:
+            raw_pty = self.cv_SYSTEM.console.get_console()
+            raw_pty.sendline(cmd)
+            raw_pty.expect("login:", timeout=600)
+            log.info("{} passed(SUCCESS)".format(cmd))
+        except CommandFailed:
+            log.info("{} failed with exception.".format(cmd))
+
+    def kexec_load(self):
+        """
+        This function will test the kexec load
+        """
+        try:
+            self.c.run_command("kexec -l /boot/{} --initrd /boot/{} --append=\"`cat /proc/cmdline`\"".format(self.kernel_image,self.initrd_image))
+        except CommandFailed:
+            kexec_loaded_status = self.c.run_command("cat /sys/kernel/kexec_loaded")
+            if kexec_loaded_status[0] == "0":
+                log.info("Failed to load kernel image = \"%s\"", self.kernel_image)
+            raise OpTestError("kexec -l failed to load kernel")
+        
+    def kexec_unload(self):
+        """
+        This function will test the kexec unload
+        """
+        try:
+            self.c.run_command("kexec -u")
+        except CommandFailed:
+            kexec_loaded_status = self.c.run_command("cat /sys/kernel/kexec_loaded")
+            if kexec_loaded_status[0] == "1":
+                log.info("Failed to unload kernel image = \"%s\"", self.kernel_image)
+            raise OpTestError("Kexec -u  failed to unload")
+        
+    def kexec_first_load_then_exec(self):
+        """
+        This function first tests kexec load and then exec(-e) 
+        """
+        self.kexec_load()
+        cmd="kexec -e"
+        self.get_raw_pty_console(cmd)
+
+
+    def kexec_force(self):
+        """
+        This function tests the kexec force(-f)
+        """
+        cmd="kexec -f /boot/{} --initrd /boot/{} --append=\"`cat /proc/cmdline`\"".format(self.kernel_image,self.initrd_image)
+        self.get_raw_pty_console(cmd)
+
+    def kexec_load_and_exec(self):
+        """
+        This function tests the kexec -l and -e together
+        """
+        cmd="kexec -l -e /boot/{} --initrd /boot/{} --append=\"`cat /proc/cmdline`\"".format(self.kernel_image,self.initrd_image)
+        self.get_raw_pty_console(cmd)
+
+    def kexec_multiboot(self):
+        """
+        This function tests the kexec multi-boot in loop of 10 iterations 
+        """
+        log.info("Start loop of 10 iterations with kexec -l -e together")
+        for i in range(10):
```

Comment:
This can be a configurable option via confirmation file.

Reply:
Added configurable option  "num_of_iterations"